tokio-current-thread/src/lib.rs

```diff
@@ -103,24 +104,88 @@ pub struct RunError {
     _p: (),
 }
 
+impl fmt::Display for RunError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.description())
+    }
+}
+
+impl Error for RunError {
+    fn description(&self) -> &str {
+        "Run error"
+    }
+    fn cause(&self) -> Option<&Error> {
+        None
+    }
+}
+
 /// Error returned by the `run_timeout` function.
 #[derive(Debug)]
 pub struct RunTimeoutError {
     timeout: bool,
 }
 
+impl fmt::Display for RunTimeoutError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.description())
+    }
+}
+
+impl Error for RunTimeoutError {
+    fn description(&self) -> &str {
+        if self.timeout {
+            "Run timeout error (timeout)"
+        } else {
+            "Run timeout error (not timeout)"
+        }
+    }
+    fn cause(&self) -> Option<&Error> {
+        None
+    }
+}
+
 /// Error returned by the `turn` function.
 #[derive(Debug)]
 pub struct TurnError {
     _p: (),
 }
 
+impl fmt::Display for TurnError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.description())
+    }
+}
+
+impl Error for TurnError {
+    fn description(&self) -> &str {
+        "Turn error"
+    }
+    fn cause(&self) -> Option<&Error> {
+        None
+    }
+}
+
 /// Error returned by the `block_on` function.
 #[derive(Debug)]
 pub struct BlockError<T> {
     inner: Option<T>,
 }
 
+impl<T> fmt::Display for BlockError<T> {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "Block error")
+    }
+}
+
+impl<T: fmt::Debug> Error for BlockError<T> {
+    fn description(&self) -> &str {
+        "Block error"
+    }
+    fn cause(&self) -> Option<&Error> {
```

Comment:
Do you really need to provide these `cause` functions?

Reply:
Ah, no, I don't (I thought both were mandatory, but they are both provided in fact).

I'd still like to keep the one in the wrapper `RunError` ‒ as it delegates to the inner error, this is more robust for the future if the inner error might change the implementation.